Sources/LaunchItem.swift

```diff
@@ -22,48 +22,69 @@ import Foundation
 
 
 public enum LaunchItem: CustomStringConvertible, Equatable {
-    case RemoteNotificationItem(remoteNotification: RemoteNotification)
-    case LocalNotificationItem(localNotification: UILocalNotification)
-    case OpenURLItem(urlToOpen: URLToOpen)
+    case remoteNotification(item: RemoteNotification)
+    case localNotification(item: UILocalNotification)
+    case openURL(item: URLToOpen)
     @available(iOS 9.0, *)
-    case ShortcutItem(shortcutItem: UIApplicationShortcutItem)
-    case UserActivityItem(userActivity: NSUserActivity)
-    case NoItem
+    case shortcut(item: UIApplicationShortcutItem)
+    case userActivity(item: NSUserActivity)
+    case none
+    
+    // MARK: Equatable
+    
+    public static func ==(lhs: LaunchItem, rhs: LaunchItem) -> Bool {
+        switch (lhs, rhs) {
+        case let (.remoteNotification(itemLHS), .remoteNotification(itemRHS)) where itemLHS == itemRHS:
+            return true
+        case let (.localNotification(itemLHS), .localNotification(itemRHS)) where itemLHS == itemRHS:
+            return true
+        case let (.openURL(itemLHS), .openURL(itemRHS)) where itemLHS == itemRHS:
+            return true
+        case let (.shortcut(itemLHS), .shortcut(itemRHS)) where itemLHS == itemRHS:
+            return true
+        case let (.userActivity(itemLHS), .userActivity(itemRHS)) where itemLHS == itemRHS:
+            return true
+        case (.none, .none):
+            return true
+        default:
+            return false
+        }
+    }
     
     // MARK: Initialization
     
-    init(launchOptions: [NSObject : AnyObject]?) {
-        if let launchRemoteNotification = RemoteNotification(remoteNotification: launchOptions?[UIApplicationLaunchOptionsRemoteNotificationKey] as? [NSObject : AnyObject]) {
-            self = .RemoteNotificationItem(remoteNotification: launchRemoteNotification)
-        } else if let launchLocalNotification = launchOptions?[UIApplicationLaunchOptionsLocalNotificationKey] as? UILocalNotification {
-            self = .LocalNotificationItem(localNotification: launchLocalNotification)
-        } else if let launchURL = launchOptions?[UIApplicationLaunchOptionsURLKey] as? NSURL {
-            let sourceApplicationBundleID = launchOptions?[UIApplicationLaunchOptionsSourceApplicationKey] as? String
-            let annotation = launchOptions?[UIApplicationLaunchOptionsAnnotationKey]
+    init(launchOptions: [AnyHashable : Any]?) {
```

Comment:
It was also already here before. I think a cleanup here is out of scope for a move to Swift 3. I will add whitespace after each case though since this is indeed a bit dense. We can look at cleaning up the implementation at a later date